### PR TITLE
Stop arrow from runtiming

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Title

## Why It's Good For The Game

There was no hardel protection

## Changelog
:cl:
fix: CIC order and rally no longer runtime
/:cl:

### CODE REVIEW

File: code/_onclick/hud/screen_objects/screen_objects.dm [hunk 1]
```diff
@@ -764,12 +764,17 @@
...
 	tracker = tracker_input
 	target = target_input
 	tracker.client.screen += src
+	RegisterSignal(tracker, COMSIG_PARENT_QDELETING, .proc/kill_arrow)
+	RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/kill_arrow)
 	process() //Ping immediately after parameters have been set
 
+///Stop the arrow to avoid runtime and hard del
+/obj/screen/arrow/proc/kill_arrow()
+	qdel(src)
```
Comment (SplinterGP) [inline code/_onclick/hud/screen_objects/screen_objects.dm:776]:
```suggestion
/obj/screen/arrow/proc/kill_arrow()
	SIGNAL_HANDLER
	qdel(src)
```

File: code/_onclick/hud/screen_objects/screen_objects.dm [hunk 2]
```diff
@@ -764,12 +764,17 @@
 /obj/screen/arrow/proc/add_hud(mob/living/carbon/tracker_input, atom/target_input)
 	if(!tracker_input?.client)
 		return
-
 	tracker = tracker_input
 	target = target_input
 	tracker.client.screen += src
```
Comment (SplinterGP) [inline code/_onclick/hud/screen_objects/screen_objects.dm:769]: probably should harddel protect this reference specifically

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/_onclick/hud/screen_objects/screen_objects.dm (hunk 2 above)
Comment (BraveMole) [reply code/_onclick/hud/screen_objects/screen_objects.dm:769]: There was literraly a runtime here splinter. So it is hapenning.
"what probably happens is the arrow getting destroyed and it not being removed from the client screen." 
I have another PR for that

**Review by TiviPlus — APPROVED** [review]

**Review by SplinterGP — CHANGES_REQUESTED** [review]
My other comment dindt go for some reason(this one)

File: code/_onclick/hud/screen_objects/screen_objects.dm [hunk 3]
```diff
@@ -784,6 +789,8 @@
 		transform = turn(transform, Get_Angle(tracker, target))
 
 /obj/screen/arrow/Destroy()
+	target = null
+	tracker = null
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/_onclick/hud/screen_objects/screen_objects.dm:794]: Vars arent anulled on destroy, they still hold refs

**Pull request merged**